src/lib/bootloader/proposal_client.rb

```diff
@@ -227,5 +222,13 @@ def single_click_action(option, value)
 
       Yast::Bootloader.proposed_cfg_changed = true
     end
+
+    def update_required_packages
+      bl = ::Bootloader::BootloaderFactory.current
+      bootloader_resolvables = Yast::PackagesProposal.GetResolvables("yast2-bootloader", :package)
+      log.info "proposed packages to install #{bl.packages}"
+      Yast::PackagesProposal.RemoveResolvables("yast2-bootloader", :package, bootloader_resolvables)
```

Comment:
Instead of `RemoveResolvables(old)` + `AddResolvables(new)` you could use one `SetResolvables(new)` which replaces the current setting. See https://github.com/yast/yast-yast2/blob/master/library/general/src/modules/PackagesProposal.rb#L134-L159